### PR TITLE
batched_span_select now guarantees element order in each span

### PR DESCRIPTION
**Pull request by OhadRubin**
Solves https://github.com/allenai/allennlp/issues/4498

### CODE REVIEW

**Comment by OhadRubin** [comment]
I'll take a look at [https://github.com/allenai/allennlp/blob/8a05ad43f1ff66b81dc1386903976d607da29a99/CONTRIBUTING.md](CONTRIBUTING.md), and workshop this a little...

**Comment by dirkgr** [comment]
I am excited about fixing this issue!

**Comment by dirkgr** [comment]
Is this ready for a review yet?

**Comment by OhadRubin** [comment]
Yes

File: allennlp/nn/util.py [hunk 1]
```diff
@@ -1318,12 +1317,11 @@ def batched_span_select(target: torch.Tensor, spans: torch.LongTensor) -> torch.
...
     span_mask = max_span_range_indices <= span_widths
-    raw_span_indices = span_ends - max_span_range_indices
-    # We also don't want to include span indices which are less than zero,
-    # which happens because some spans near the beginning of the sequence
-    # have an end index < max_batch_span_width, so we add this to the mask here.
-    span_mask = span_mask & (raw_span_indices >= 0)
-    span_indices = torch.nn.functional.relu(raw_span_indices.float()).long()
+    raw_span_indices = span_starts + max_span_range_indices
+    # We also don't want to include span indices which greater than sequence_length,
+    # which happens because some spans near the end of the sequence
```
Comment (dirkgr) [inline allennlp/nn/util.py:1322]: This sentence is truncated?

Comment (OhadRubin) [reply allennlp/nn/util.py:1322]: Yeah, fixed that.

**Review by dirkgr — APPROVED** [review]
Thank you!

**Pull request merged**